storage/innobase/trx/trx0purge.cc

```diff
@@ -169,19 +169,21 @@ void purge_sys_t::create()
   ut_ad(this == &purge_sys);
   ut_ad(!m_initialized);
   ut_ad(!enabled());
+  ut_ad(!m_active);
+  m_skipped_rseg= srv_undo_tablespaces > 0;
```

Comment:
I agree on the necessity of the comment. I prefer to avoid `if` whenever possible. There might be no conditional jump in the target code either; it could be `cmp` followed by `setg`, or `test` followed by `setne`. The code with `if` would hopefully compile to the same code, but it might not, especially at lower optimization levels.